docs(spin/install): updates per v1.1.0 (#515)
fermyon/developer: content/spin/install.md

<!-- @selectiveCpy -->

> <pre class="bash spin-install" id="spin-install-source-cargo"><code>$ git clone https://github.com/fermyon/spin -b v1.1.0

Reviewer: This won't work until we actually create a `v1.1.0` tag, of course.

$ cd spin
$ rustup target add wasm32-wasi
$ rustup target add wasm32-unknown-unknown
$ cargo install --locked --path .
$ spin --help
</code></pre>

> > Please note: Spin v1.1.0 requires `wasmtime v7.0.0` which requires rustc 1.68.0 or newer. You can update Rust using the following command:

Reviewer 2: Consider dropping the "reason why" sentence. No user cares which dependency induces the version requirement _grin_. I also suggest qualifying "Spin requires" lest a reader's eyes slide past the section header and they get the impression that the Spin binary depends on having Rust installed.
 
```suggestion
> Please note: Installing Spin v1.1.0 from source requires Rust 1.68.0 or newer. You can update Rust using the following command:
```
